File: C/experiencias.c
#include <stdio.h>
#include <locale.h>

int casos, quantidade, total, totalC, totalR, totalS;
char cobaia;
double percentualC, percentualR, percentualS;

int main() {
    setlocale(LC_ALL, "Portuguese");

    printf("Quantos casos de teste serão digitados? ");
    scanf("%d", &casos);

    for (int i = 0; i < casos; i++) {
        printf("Quantidade de cobaias: ");
        scanf("%d", &quantidade);
        printf("Tipo de cobaia: ");
        scanf("%s", &cobaia);

        total += quantidade;
        if (cobaia == 'C') {
            totalC += quantidade;
        }
        else if (cobaia == 'R') {
            totalR += quantidade;
        }
        else if (cobaia == 'S') {
            totalS += quantidade;
        }

    }

    percentualC = (double)totalC / total * 100;
    percentualR = (double)totalR / total * 100;
    percentualS = (double)totalS / total * 100;

    printf("\n");
    printf("RELATORIO FINAL: \n");
    printf("Total: %d cobaias \n", total);
    printf("Total de coelhos: %d \n", totalC);
    printf("Total de ratos: %d \n", totalR);
    printf("Total de sapos: %d \n", totalS);
    printf("Percentual de coelhos: %.2lf \n", percentualC);
    printf("Percentual de ratos: %.2lf \n", percentualR);
    printf("Percentual de sapos: %.2lf \n", percentualS);

}

File: Python/experiencias.py
casos = int(input("Quantos casos de teste serão digitados? "))

coelhos = 0.0; ratos = 0.0; sapos = 0.0; percentualC = 0.0; percentualR = 0.0; percentualS = 0.0; total = 0.0
for i in range(0, casos):
    quantidade = int(input("Quantidade de cobaias: "))
    tipo = input("Tipo de cobaia: ")

    if tipo == "C":
        coelhos += quantidade
    elif tipo == "R":
        ratos += quantidade
    elif tipo == "S":
        sapos += quantidade

    total += quantidade
    percentualC = (coelhos / total) * 100
    percentualR = (ratos / total) * 100
    percentualS = (sapos / total) * 100

print("RELATORIO FINAL: ")
print(f"Total: {total:.0f}")
print(f"Total de coelhos: {coelhos:.0f}")
print(f"Total de ratos: {ratos:.0f}")
print(f"Total de sapos: {sapos:.0f}")
print(f"Percentual de coelhos: {percentualC:.2f}")
print(f"Percentual de ratos: {percentualR:.2f}")
print(f"Percentual de sapos: {percentualS:.2f}")
